client/post-editor/editor-ground-control/index.jsx

```diff
@@ -141,7 +141,7 @@ export class EditorGroundControl extends React.Component {
 		// find the last non-editor path in routeHistory, default to "all posts"
 		const lastNonEditorPath = findLast(
 			this.props.routeHistory,
-			action => ! action.path.match( /^\/(post|page|(edit\/[^\/]+))\/[^\/]+(\/\d+)?$/i )
+			action => ! ( action.path + '/' ).match( /^\/(post|page|edit)\//i )
```

Comment:
You can make the trailing slash optional without changing the tested string. The improved regexp is:
```js
/^\/(post|page|edit)($|\/)/i
```
Matches `/page`, matches `/page/site.blog`, doesn't match `/pages`.

Reply:
Fully agree with your proposal. This is a better way to handle that. :thumbsup:
Ill update it.